### PR TITLE
Added licensing information to Building Clusters

### PR DESCRIPTION
**Pull request by stevenGravy**
## Description
Provide documentation on creating and updating licenses for Gravity

## Type of change
Docs

## TODOs

- [x] Self-review the change
- [x] Write tests
- [x] Perform manual testing
- [x] Write documentation
- [x] Address review feedback

## Testing done
Confirmed these license instructions work

### CODE REVIEW

File: docs/7.x/pack.md [hunk 1]
```diff
@@ -905,6 +905,69 @@ installer:
...
+
+Gravity Enterprise provides the ability to generate licenses and require a license for cluster installations.  Licenses are restrictable for the number of nodes per cluster and a license expiration date.  You have the option of requiring the application to stop when a license expires.
+
+Here are the commands available in `gravity` for creating, installing, updating and displaying the license in the cluster.  
+
+| Command        | Description |
+|----------------|-------------|
+| `gravity license new`   | Generates a new license
+| `gravity license install`    | Install (or update) a cluster license.
+| `gravity license show`    | Downloads a Cluster Image from a Gravity Hub.
```
Comment (r0mant) [inline docs/7.x/pack.md:918]: The description seems out of place. `Displays the currently installed cluster license`?

File: docs/7.x/pack.md [hunk 2]
```diff
@@ -905,6 +905,69 @@ installer:
...
+To see all the options on each command use `--help` as a paramter on each command (Ex: `gravity license new --help`)
+
+Licenses are generated from the `gravity` executable using a Certificate Authority (CA) with a private key and an optional encryption key. After generating the license the CA and optional encryption key is used in a cluster build that has licensing enabled.  Customers will then need to have a license in the CLI or wizard installation for those cluster installations.  Prior to a license expiration the `graviity install` command allows for updating the licenses in the clusters.
```
Comment (r0mant) [inline docs/7.x/pack.md:921]: Typo on "graviity", also `gravity license install`?

File: docs/7.x/pack.md [hunk 3]
```diff
@@ -905,6 +905,69 @@ installer:
...
+  ```bash
+  openssl req -newkey rsa:2048 -nodes -keyout domain.key -x509 -days 365 -out domain.crt
+  ```
+  
+  2. Generate a license with an encryption key using the CA
+  ```bash
+  gravity license new --max-nodes=3 --valid-for=8760h --ca-cert=domain.crt --ca-key=domain.key --encryption-key=833dcd744b5aed710711ca6f0f8a66b3712f63f17ef03146 > license.pem
+  ```
+  
+  3. Add license enabled to the Cluster Application YAML
```
Comment (r0mant) [inline docs/7.x/pack.md:937]: "Enable licensing feature in the cluster image manifest"

File: docs/7.x/pack.md [hunk 4]
```diff
@@ -905,6 +905,69 @@ installer:
...
+Gravity Enterprise provides the ability to generate licenses and require a license for cluster installations.  Licenses are restrictable for the number of nodes per cluster and a license expiration date.  You have the option of requiring the application to stop when a license expires.
+
+Here are the commands available in `gravity` for creating, installing, updating and displaying the license in the cluster.  
+
+| Command        | Description |
+|----------------|-------------|
+| `gravity license new`   | Generates a new license
+| `gravity license install`    | Install (or update) a cluster license.
+| `gravity license show`    | Downloads a Cluster Image from a Gravity Hub.
+To see all the options on each command use `--help` as a paramter on each command (Ex: `gravity license new --help`)
```
Comment (a-palchikov) [inline docs/7.x/pack.md:919]: parameter
To see usage details for each command, append `--help` (Ex. `gravity license new --help`)

File: docs/7.x/pack.md (hunk 2 above)
Comment (a-palchikov) [inline docs/7.x/pack.md:921]: After generating the license **,** the CA and ...

File: docs/7.x/pack.md [hunk 5]
```diff
@@ -905,6 +905,69 @@ installer:
...
+apiVersion: cluster.gravitational.io/v2
+kind: Cluster
+metadata:
+    name: Sample
+    resourceVersion: 1.0.0
+license:
+    enabled: true
+  ```
+  
+  4. Build the cluster with the CA and encryption key
```
Comment (a-palchikov) [inline docs/7.x/pack.md:949]: and the same encryption key

File: docs/7.x/pack.md [hunk 6]
```diff
@@ -905,6 +905,69 @@ installer:
...
+  ```bash
+  tele build app.yaml --ca-cert=domain.crt --encryption-key=833dcd744b5aed710711ca6f0f8a66b3712f63f17ef03146
+  ```
+  
+  5. Install with the license file
+  ```bash
+  sudo ./gravity install --license-file=/tmp/license.pem
+  ```
+
+Repeat steps 1 and 2 at a later date to generate new licenses.  To install a new license file into a cluster use this command.
```
Comment (a-palchikov) [inline docs/7.x/pack.md:960]: ... use this command:

**Comment by stevenGravy** [comment]
@r0mant  @a-palchikov  thanks for reviewing.  I have addressed the feedback

File: docs/7.x/pack.md [hunk 7]
```diff
@@ -905,6 +905,69 @@ installer:
...
+
+Gravity Enterprise provides the ability to generate licenses and require a license for cluster installations.  Licenses are restrictable for the number of nodes per cluster and a license expiration date.  You have the option of requiring the application to stop when a license expires.
+
+Here are the commands available in `gravity` for creating, installing, updating and displaying the license in the cluster.  
+
+| Command        | Description |
+|----------------|-------------|
+| `gravity license new`   | Generates a new license.
+| `gravity license install`    | Install (or update) a cluster license.
+| `gravity license show`    | Downloads a Cluster Image from a Gravity Hub.
```
Comment (a-palchikov) [inline docs/7.x/pack.md:918]: The comment is still out of place though.

**Review by a-palchikov — APPROVED** [review]
lgtm with one remark

File: docs/7.x/pack.md (hunk 7 above)
Comment (stevenGravy) [reply docs/7.x/pack.md]: @a-palchikov  sorry about that, fixed.

**Pull request merged**